bokeh/plotting/helpers.py

```diff
@@ -882,31 +926,31 @@ def func(self, **kwargs):
             kwargs['global_alpha'] = kwargs['alpha']
```

Comment:
@solstag it was happening here, presumably

Reply:
To be clear I'm not sure why this isn't still working since it seems `_pop_visuals` is called later... all that said this is pretty hacky so I'd be open to any changes that keep the behavior in a less hacky way

Reply:
I see.

Well, it doesn't work with _pop_visuals for the reason I mentioned. The parameter that is having an effect on the glyph is not `alpha`, but `global_alpha`.

The parameter `alpha` is being treated the way I mentioned, with `_pop_colors_and_alpha` incorrectly consuming it but doing nothing, and `_pop_visuals` not consuming it since it has nothing to do. However since the glyph doesn't have `alpha`, the correct behaviour will yield an error later on.

At the same time, the parameter `global_alpha` is ignored by both, and treated downstream. Thus it is only circumstantial that `_pop_colors_and_alpha` yields the expected result, since as as you noticed the code hacks the value of "alpha" into "global_alpha" (I missed that one).

So in my opinion we have two reasonable choices:
- treat "global" as a visual feature so that trait aliases will work as usual: not only for Image{,RGBA,URL} and not only for `_alpha`.
- choose between "global_alpha" or "alpha" and adapt the Image glyphs accordingly.

Not knowing the general usage of `global_` in bokeh, if I had to choose it by myself I would go for the second choice, keeping the most used variation. If designing from scratch I would have chosen something more meaningful like "image_alpha", as "image" would naturally go a "visual" feature.

Does that seem right? What do you think?

Reply:
I think we have to keep both, unfortunately. The reason the model property is called `global_alpha` is because the image data itself an contain an alpha channel. This is to be clear that the alpha applies globally and additively on top of any alpha in the data. The `alpha` for the glyph *functions* is a convenience but at this point, a well-established one and changing it would break many people I suspect. I think this just has to be handled as a special case for the relevant glyphs. If we want to be fancy we can implement some general way to specify per-glyph aliases like this and then configure it appropriately. But I am also OK at this point with just putting a few lines of special case code `pop_visuals` to handle images.

Reply:
I don't like making "global" a visual feature because as you say:

> not only for Image{,RGBA,URL} and not only for _alpha.

But those are truly the only places they apply (and should apply)

Reply:
Hm... can't this be handled by the glyph itself? Exposing two properties that have the same semantics is not really a bad thing modulo documentation, and it would make something that is glyph specific consistent when handling the glyph with or without the plotting API.

Otherwise we'll end up with a "plotting API specific alpha" that does not respect the behaviour of "alpha in the plotting API". Let alone if `Image*` glyphs ever get a "[line|text]_alpha" or a "border_alpha", either in the future or if someone subclasses those in their application.

It doesn't seem nice to expose users to behaviour dependent on the product of glyphs and APIs. «If something is glyph specific it works with every API, if something is API specific it works with every glyph.» is a nice UX principle, and it seems to me (may be wrong as I am new to bokeh devel) that we're pretty close to that as far as the plotting API is concerned.

Reply:
>  if Image* glyphs ever get a [line|text]_alpha or a border_alpha

They won't

> or if someone subclasses those in their application.

Just being honest: I don't care about this case (which also is very unlikely to actually happen)

Here is the thing to note: None of the glyphs have `alpha`. Not one. `alpha` is purely a convenience layer added at the `bokeh.plotting` layer. Making sure that `alpha` works reasonably (i.e. does the thing users will expect) for *all* the glyphs is exactly the most user-friendly thing to do. In the case of other glyphs, setting `alpha` is a convenience for setting both `line_alpha` and `fill_alpha`. That's no different from having it set `global_alpha` for images (because, for images, that is the reasonable thing users will expect to work). 

This snot a documentation problem, because making it work for everything means:

> Setting `alpha` in a glyph method is a shortcut for setting all the relevant "alpha"-like properties of a glyph

Contrast if we *don't* do that:

> Setting `alpha` in a glyph method is a shortcut for setting all the relevant "alpha"-like properties of a glyph, except for all these glyphs where that doesn't apply: ..., ...

That's worse for documentation.

Reply:
I will say, if you are only willing to handle this by adding `global` as another "visual" than that's your call and I would accept the PR. But we need to have a *property* "global_alpha" that is aliased by "alpha" in `bokeh.plotting` glyph methods, however that is accomplished.